src/pages/TokenDetails/index.tsx

```diff
@@ -35,8 +39,14 @@ export default function TokenDetails() {
   const pageChainId = CHAIN_NAME_TO_CHAIN_ID[chain]
   const nativeCurrency = nativeOnChain(pageChainId)
   const isNative = tokenAddress === NATIVE_CHAIN_ID
-  const tokenQueryData = useTokenQuery(isNative ? nativeCurrency.wrapped.address : tokenAddress ?? '', chain)
-  const prices = useTokenPriceQuery(isNative ? nativeCurrency.wrapped.address : tokenAddress ?? '', chain)
+  const timePeriod = useAtomValue(filterTimeAtom)
+  const tokenQueryReference = useLoadTokenQuery(isNative ? nativeCurrency.wrapped.address : tokenAddress ?? '', chain)
+  const priceQueryReference = useLoadTokenPriceQuery(
+    isNative ? nativeCurrency.wrapped.address : tokenAddress ?? '',
+    chain,
+    timePeriod
+  )
+  const tokenQueryData = usePreloadedQuery<TokenQuery>(tokenQuery, tokenQueryReference).tokens?.[0]
```

Comment:
Because you are using the preloaded query immediately after triggering, you should just useLazyLoadQuery.